src/App.js

```diff
@@ -9,36 +9,46 @@ class App extends Component {
       { id: 2, value: 0 },
       { id: 3, value: 0 },
       { id: 4, value: 0 }
-    ]
+    ],
+    total: 0
   };
 
   handleIncrement = counter => {
     const counters = [...this.state.counters];
     const index = counters.indexOf(counter);
     counters[index] = { ...counters[index] };
     counters[index].value++;
-    this.setState({ counters });
+    let total = this.state.total;
+    total++;
+    this.setState({ counters, total });
   };
 
   handleDecrement = counter => {
     const counters = [...this.state.counters];
     const index = counters.indexOf(counter);
     counters[index] = { ...counters[index] };
     counters[index].value--;
-    this.setState({ counters });
+    let total = this.state.total;
+    total--;
+    this.setState({ counters, total });
   };
 
   handleReset = () => {
     const counters = this.state.counters.map(c => {
       c.value = 0;
       return c;
     });
-    this.setState({ counters });
+    let total = 0;
+    this.setState({ counters, total });
   };
 
   handleDelete = counterId => {
     const counters = this.state.counters.filter(c => c.id !== counterId);
-    this.setState({ counters });
+    let total = 0;
+    counters.forEach(c => {
+      total = total + c.value;
+    })
```

Comment:
We can use reduce method also.
`const total = arr.counters((acc,el)=>acc+el.value,0);`